src/utilstrencodings.cpp

```diff
@@ -123,6 +125,75 @@ vector<unsigned char> ParseHex(const string& str)
     return ParseHex(str.c_str());
 }
 
+string EncodeAscii85(const char* istr, size_t len)
+{
+    
+    string retVal; //Default is empty-string
+    if (NULL != istr)
+    {
+        size_t isz = strlen(istr);
+        int32_t olen = ascii85_get_max_encoded_length(isz);
+        
+        if (olen < 0)
+        {
+            return retVal; //Encode size error
+        }
+        else
+        {
+            uint8_t obuf[olen];
+            olen = encode_ascii85((uint8_t *)istr, isz, obuf, olen);
```

Comment:
don't use c-style casts in cpp code. 
reinterpret_cast<const uint8_t *>(istr), and don't forget about const.